javaparser-metamodel-generator/src/main/java/com/github/javaparser/generator/metamodel/MetaModelGenerator.java

```diff
@@ -0,0 +1,269 @@
+        add(BooleanLiteralExpr.class);
+        add(CastExpr.class);
+        add(CharLiteralExpr.class);
+        add(ClassExpr.class);
+        add(ConditionalExpr.class);
+        add(DoubleLiteralExpr.class);
+        add(EnclosedExpr.class);
+        add(FieldAccessExpr.class);
+        add(InstanceOfExpr.class);
+        add(IntegerLiteralExpr.class);
+        add(LambdaExpr.class);
+        add(LongLiteralExpr.class);
+        add(MarkerAnnotationExpr.class);
+        add(MemberValuePair.class);
+        add(MethodCallExpr.class);
+        add(MethodReferenceExpr.class);
+        add(NameExpr.class);
+        add(Name.class);
+        add(NormalAnnotationExpr.class);
+        add(NullLiteralExpr.class);
+        add(ObjectCreationExpr.class);
+        add(SimpleName.class);
+        add(SingleMemberAnnotationExpr.class);
+        add(SuperExpr.class);
+        add(ThisExpr.class);
+        add(TypeExpr.class);
+        add(UnaryExpr.class);
+        add(VariableDeclarationExpr.class);
+
+        add(ImportDeclaration.class);
+
+        add(AssertStmt.class);
+        add(BlockStmt.class);
+        add(BreakStmt.class);
+        add(CatchClause.class);
+        add(ContinueStmt.class);
+        add(DoStmt.class);
+        add(EmptyStmt.class);
+        add(ExplicitConstructorInvocationStmt.class);
+        add(ExpressionStmt.class);
+        add(ForeachStmt.class);
+        add(ForStmt.class);
+        add(IfStmt.class);
+        add(LabeledStmt.class);
+        add(ReturnStmt.class);
+        add(SwitchEntryStmt.class);
+        add(SwitchStmt.class);
+        add(SynchronizedStmt.class);
+        add(ThrowStmt.class);
+        add(TryStmt.class);
+        add(LocalClassDeclarationStmt.class);
+        add(WhileStmt.class);
+
+        add(ArrayType.class);
+        add(ClassOrInterfaceType.class);
+        add(IntersectionType.class);
+        add(PrimitiveType.class);
+        add(TypeParameter.class);
+        add(UnionType.class);
+        add(UnknownType.class);
+        add(VoidType.class);
+        add(WildcardType.class);
+    }};
+
+    private static String METAMODEL_PACKAGE = "com.github.javaparser.metamodel";
+
+    public static void main(String[] args) throws IOException, NoSuchMethodException {
+        new MetaModelGenerator().run();
+    }
+
+    private void run() throws IOException, NoSuchMethodException {
+        final Path root = getJavaParserBasePath().resolve(Paths.get("javaparser-metamodel", "src", "main", "java"));
+
+        JavaParser javaParser = new JavaParser();
+
+        SourceRoot sourceRoot = new SourceRoot(root);
+
+        CompilationUnit javaParserMetaModel = sourceRoot.parse(METAMODEL_PACKAGE, "JavaParserMetaModel.java", javaParser).get();
+
+        generateClassMetaModels(javaParserMetaModel, sourceRoot);
+
+        sourceRoot.saveAll();
+    }
+
+    private void generateClassMetaModels(CompilationUnit javaParserMetaModelCu, SourceRoot sourceRoot) throws NoSuchMethodException {
```

Comment:
This methods seems quite heavy 30+ LOC. Maybe you are doing two different things and it is splitable ? :)